presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -510,51 +523,52 @@ private void assertRoundTrip(Type writeType, Type readType, List<?> writeValues,
     }
 
     private static void assertFileContentsPresto(
-            Type type,
+            List<Type> types,
             TempFile tempFile,
-            List<?> expectedValues,
+            List<List<?>> expectedValues,
             boolean skipFirstBatch,
             boolean skipStripe,
             OrcEncoding orcEncoding,
             Format format,
             boolean isHiveWriter)
             throws IOException
     {
-        try (OrcRecordReader recordReader = createCustomOrcRecordReader(tempFile, orcEncoding, createOrcPredicate(type, expectedValues, format, isHiveWriter), type, MAX_BATCH_SIZE)) {
+        OrcPredicate orcPredicate = types.size() > 1 ? TRUE : createOrcPredicate(types.get(0), expectedValues.get(0), format, isHiveWriter);
```

Comment:
I think instead of skipping the rest of the predicates, there is a quite fix to `TestingOrcPredicate` to enforce predicates for all columns. All predicate classes in `TestingOrcPredicate` inherit `BasicOrcPredicate`. Instead of having `List<T> expectedValues` in `BasicOrcPredicate`, we could make it a list of list like what you have changed for `OrcTester` class. The input `statisticsByColumnIndex` should have the exact same size as `expectedValues`. We can assert column stats one after another.

Reply:
@highker You are right. Looks like it is quite straightforward to add support for multiple columns to TestingOrcPredicate. I put the changes into a fixup commit hoping it would be easier to review: https://github.com/prestodb/presto/pull/12991/commits/15c5f09328db769aae58937d25fcaa244195dade